stdlib/public/core/SwiftNativeNSArray.swift

```diff
@@ -124,26 +139,34 @@ extension _SwiftNativeNSArrayWithContiguousStorage : _NSArrayCore {
 ///
 /// Ideally instances of this class would be allocated in-line in the
 /// buffers used for Array storage.
+@_fixed_layout
```

Comment:
We should measure the overload of having these classes not be fixed layout (which means the ctor cannot be inlined). Maybe inlining the allocating entry point (which should always be safe) will make up for it :)